scan.go

```diff
@@ -221,12 +242,29 @@ func (s *ScanService) Do() (*ScanCursor, error) {
 		path += "/" + strings.Join(typesPart, ",")
 	}
 
+	// Define the body of the request
+	var body map[string]interface{}
+
+	if s.source != nil {
+		body = s.source
+	} else {
+		src, err := s.searchSource.Source()
+
+		if err != nil {
+			return nil, err
+		}
+
+		// searchSource.Source() actually returns a map, yet its return type
+		// is specified as interface...
```

Comment:
I explicitly stated that in the documentation of interfaces, e.g. [`Query`](https://github.com/olivere/elastic/blob/release-branch.v3/query.go). I know no way of enforcing it with the compiler.

I agree that documentation could be better throughout the library.

You're correct in that I should have hidden many things; unfortunately, it's about 3 years too late for that.